cartographer/io/internal/mapping_state_serialization.cc

```diff
@@ -83,15 +83,13 @@ void SerializeSubmaps(
   SerializedData proto;
```

Comment:
Style: Place a function's variables in the narrowest scope possible, and initialize variables in the declaration.

Can we please move this variable and similar variables in other functions below into the loop? I doubt there is any measurable benefit and reusing the variable harms readability. It's not needed outside the inner scope.

If there were any advantage and this intentional, there should be a comment explaining this otherwise surprising and non-style-conforming code structure.